comit/src/network/swap_digest.rs

```diff
@@ -1,19 +1,42 @@
 use crate::{asset, identity, network::protocols::announce::SwapDigest, RelativeTime, Timestamp};
-use digest::Digest;
+use digest::{Digest, ToDigestInput};
+
+// TODO should we use swap_protocols::SwapProtocol instead?
+#[derive(Clone, Digest, Debug, PartialEq, Copy)]
+#[digest(hash = "SwapDigest")]
+pub enum SwapProtocol {
+    #[digest(prefix = "4b17")]
+    Hbit,
+    #[digest(prefix = "e5ec")]
+    Herc20,
+    #[digest(prefix = "c3d3")]
+    Halight,
+}
+
+// TODO why is this necessary?
+impl ToDigestInput for SwapProtocol {
+    fn to_digest_input(&self) -> Vec<u8> {
+        self.clone().digest().to_digest_input()
+    }
+}
 
```

Comment:
You can remove this in favour of the suggestion above and _do not_ derive digest on the Protocol enum.

Reply:
Yeah, `Digest` should only be derived on the root struct. Otherwise you are doing the algorithm several times, nested in each other.